From 8b2802d1c3ea6fade5fed9be644ef4210ad07482 Mon Sep 17 00:00:00 2001
From: Ruikai Liu <rickliu2000@outlook.com>
Date: Thu, 20 Aug 2026 16:50:36 +0800
Subject: [PATCH] use CONFIG mode for protobuf instead of REQUIRED mode

CMake's FindProtobuf module and Protobuf's config both define
protobuf targets and gRPCConfig.cmake finds protobuf config,
which is causing conflict and breaks configure
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a93a38d7501..dd1f466f34f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1026,7 +1026,7 @@ if(WITH_NVMEOF_GATEWAY_MONITOR_CLIENT)
   # Find Protobuf installation
   # Looks for protobuf-config.cmake file installed by Protobuf's cmake installation.
   option(protobuf_MODULE_COMPATIBLE TRUE)
-  find_package(Protobuf REQUIRED)
+  find_package(Protobuf CONFIG REQUIRED)
 
   set(_REFLECTION grpc++_reflection)
   if(CMAKE_CROSSCOMPILING)
-- 
2.55.0

